processor/src/memory/tests.rs

```diff
@@ -50,10 +44,12 @@ fn mem_read() {
     assert_eq!(4, mem.trace_len());
 
     // check generated trace; rows should be sorted by address and then clock cycle
-    let MemoryTrace {
-        trace,
-        range_checks,
-    } = mem.into_trace();
+    let num_rows = 4;
+    let mut trace = (0..14)
+        .map(|_| vec![Felt::ZERO; num_rows])
+        .collect::<Vec<_>>();
+    let mut fragment = TraceFragment::trace_to_fragment(&mut trace);
+    mem.fill_trace(&mut fragment);
```

Comment:
Seems like this is repeated a few times. Is there a good way to put it into a helper functions (e.g., `build_trace`)?

Reply:
Yeah, happy to do so. I actually just restored this exactly to the previous version, now that we've gone back to using `fill_trace`